### ISSUE TITLE
Different ordering of quadrature points and dofs in 1d

### ISSUE THREAD

**Issue by IgorBaratta** [FEniCS/basix#352]
On spectral/collocation methods we want dofs and quadrature points to be collocated. This is possible now with a carefull selection of Lagrange variant and quadrature rule.

However dofs and quadrature points on the reference interval are numbered differently:

[Top: quadrature points,  Bottom: dofs]
![MicrosoftTeams-image](https://user-images.githubusercontent.com/15614155/144250554-854bb56d-6181-448f-951f-50f6d8c30917.png)

So it's not currently possible to implement withou  indirect memory access patterns in the kernels.

**Proposed solution:**
Reorder quadrature points and weights so they match the degrees of freedom.

**Comment by IgorBaratta** [comment]
Simple test:

```python
import basix

P = 2
Q = 3

cell_type = basix.CellType.interval
element = basix.create_element(basix.ElementFamily.P, cell_type, P, basix.LagrangeVariant.gll_warped)
points, w = basix.make_quadrature(basix.QuadratureType.gll, cell_type, Q)

phi = element.tabulate(0, points)[0, :, :, 0]
phi[phi<1e-9] = 0

# FAIL
assert numpy.count_nonzero(phi - numpy.diag(numpy.diagonal(phi))) == 0

```

**Comment by adeebkor** [comment]
Resolves in #355
